compose/config/config.py

```diff
@@ -541,12 +541,13 @@ def path_mappings_from_dict(d):
     return [join_path_mapping(v) for v in d.items()]
 
 
-def split_path_mapping(string):
-    if ':' in string:
-        (host, container) = string.split(':', 1)
-        return (container, host)
+def split_path_mapping(volume_path):
+    drive, volume_config = os.path.splitdrive(volume_path)
+    if ':' in volume_config:
+        (host, container) = volume_config.split(':', 1)
+        return (container, drive + host)
     else:
-        return (string, None)
+        return (volume_path, None)
```

Comment:
We stripped the drive from `volume_path`, don't we need to re-add it?

I guess in this case it should always be the empty string, since the container path can't be a windows path?

This is really just a question, not a blocking issue.